### ISSUE TITLE
[Feature Request] how to reduce delay between actions globally?

### ISSUE THREAD

**Issue by DanteAndroid** [mobile-dev-inc/maestro#1528]
It seems that the delay is set to a very long value, which is about 1-2 seconds after ui is fully drawn. I would like to reduce the delay between actions globally, to get faster test speed in every flow.

**Comment by arildojr7** [comment]
Yeah, there are some `waitForAppToSettle` on code that make it slower. On a local build, I decreased its value and worked fine and faster. I'm sure it was introduced for some reason, but maybe there's another approach, like taking screenshots internally and comparing them (if the screenshoots are the same in a 300ms interval, we can assume that app is settle...)

**Comment by DanteAndroid** [comment]
> Yeah, there are some `waitForAppToSettle` on code that make it slower. On a local build, I decreased its value and worked fine and faster. I'm sure it was introduced for some reason, but maybe there's another approach, like taking screenshots internally and comparing them (if the screenshoots are the same in a 300ms interval, we can assume that app is settle...)

If so I would like to know the reason. And your approach is inspiring!

**Comment by CyberCyclone** [comment]
The time between tests is a real pain point (3+ seconds!). The example video on the home page seems like it was artificially sped up. E.g, Tap events I want to be near instant and if I want to delay the next action, there's already `assertVisible` or `assertNotVisible` that can be used to delay actions. Once the conditions of those are met, then I want to immediately trigger the next action.

It would be nice to be able to configure this on a global setting if there's an action I want to deliberately wait, then there should be a `waitTimeout` as an action.

**Comment by ice-cap0** [comment]
yeah this would be great

**Comment by EvanPartidas** [comment]
> Yeah, there are some `waitForAppToSettle` on code that make it slower. On a local build, I decreased its value and worked fine and faster. I'm sure it was introduced for some reason, but maybe there's another approach, like taking screenshots internally and comparing them (if the screenshoots are the same in a 300ms interval, we can assume that app is settle...)

Are you able to decrease total delay between tapOn commands to something along the lines of 100-250ms?

Even when setting `waitToSettleTimeoutMs` to a very low number I can't get the tapOn commands to be fast enough to beat a count-down timer in my app. This pitfall sadly makes maestro unusable for testing vital flows within my app. Not sure why it takes multiple seconds for Maestro to locate and tap on an element. [Even this chimpanzee is way quicker than Maestro](https://www.youtube.com/watch?v=PNrWUS13th8&ab_channel=CellPress)

**Comment by Egehanozsoy** [comment]
Using Maestro for the first time, came directly here. It is extremely slow but the developer experience is great. Would love this to be fixed!

**Comment by Erick-Bueno** [comment]
Man, this would really be a huge improvement, the delay between actions is really annoying, I'm looking forward to this being better because the tool is really good.

**Comment by Fishbowler** [comment]
1.41 improved performance for iOS.

Anyone got concrete examples of slow executions that they can share?

**Comment by Erick-Bueno** [comment]
> 1.41 improved performance for iOS.
> 
> Anyone got concrete examples of slow executions that they can share?

The time between actions in the tests (like selecting a field, typing, etc.) is very long. This makes the tests take a lot of time to run. At my company, with more than 300 tests, this delay becomes a big problem. I want to know if there’s a way to configure the speed or reduce the delay between interactions to make the tests run faster.

**Comment by Fishbowler** [comment]
That's not a configuration option, no.

If you come join the community Slack, I'd be interested in chatting through your specifics.

**Comment by Erick-Bueno** [comment]
> That's not a configuration option, no.
> 
> If you come join the community Slack, I'd be interested in chatting through your specifics.

sure, how can i join?

**Comment by JoeBoulton** [comment]
The 1.41 update does make things noticable quicker for iOS, still not on the same level as something like playwright.

Quite slow on Android however, would be great to be able to reduce the delay between actions

**Comment by Fishbowler** [comment]
I don't think Playwright is a sensible comparison when considering mobile app testing - it's a different tool for a different job. It's not dependent on the operating system APIs, and can exert a higher degree of control.

That's not to say that Android couldn't be quicker, mind you.

**Comment by lukemmtt** [comment]
+1; shorter (or configurable) duration between actions is essential for my use case

**Comment by pai-vik** [comment]
Delay between actions is too high on Android, makes debugging extremely painful

**Comment by Fishbowler** [comment]
@lukemmtt @basevik 

You're understanding that Maestro isn't intentionally asleep during this time, right? There's probably very few folks out there building intentionally slower software.

Anyone know kotlin and want to take a peek at this slowness?

**Comment by cvipul** [comment]
I can try to take a look, any pointers on where we should look?

**Comment by Fishbowler** [comment]
AndroidDriver and AndroidDevice, anything that looks like waitToSettle

**Comment by svenMolhuijsen** [comment]
This issue is driving me nuts and im now even considering using Patrol + Playwright for the flutter application i develop. would take way more time to develop, but works way faster

**Comment by Fishbowler** [comment]
@svenMolhuijsen Have you taken any look at why your tests are slower?

One of the reasons I love Maestro is the open source nature. Not only can I know exactly how it works, but I can move to something else if I fancy (and given the open source nature, I might even be able to write a tool that would perfectly migrate me).

There's some pointers above on how you might go about engaging with the source. We've also got an active Slack community for more active discussion, if this is something you want help with.

**Comment by Fishbowler** [comment]
I've not used Patrol. That's come out since last time I was engaged in the Flutter ecosystem. I've done a quick read on it, and those tests are operating at a very different level. 

If I worked in a shop doing Flutter, I'd probably want both. Patrol tests can touch app state that Maestro wouldn't. Maestro has control over an entire device that Patrol can't.
